packages/truffle-core/lib/debug/compiler.js

```diff
@@ -15,19 +15,9 @@ class DebugCompiler {
     ); //clone
     compileConfig.quiet = true;
 
-    // since `compile.all()` returns two results, we can't use promisify
-    // and are instead stuck with using an explicit Promise constructor
-    const { contracts, files } = await new Promise((accept, reject) => {
-      compile.all(compileConfig, (err, contracts, files) => {
-        if (err) {
-          return reject(err);
-        }
+    const { contracts, sourceIndexes } = await compile.all(compileConfig);
```

Comment:
I'm putting this comment here since I can't put it on the lines you didn't change, but... as long as you're redoing this bit, this seems like a good time to make that change to using `config.with` that you suggested earlier instead of my manual-clone-and-alter?

(Also, "indexes"? :P )

Reply:
Recently @fainashalts discovered that `config.with` is actually a bit broken right now, so I think I'd rather leave it for now until we get to the bottom of it.

And "indexes" appears to be acceptable use here? Not sure.